api/src/main/java/io/grpc/ClientInterceptor.java

```diff
@@ -24,10 +24,14 @@
  * <p>Implementers use this mechanism to add cross-cutting behavior to {@link Channel} and
  * stub implementations. Common examples of such behavior include:
  * <ul>
- * <li>Adding credentials to header metadata</li>
  * <li>Logging and monitoring call behavior</li>
+ * <li>Adding metadata for proxies to observe</li>
```

Comment:
Metadata can be added for anything - why limit it to "for proxies to observe"?  (I know it's only an example but still...)

Reply:
These are just examples. There are other possibilities. The listed items are use-cases, not just functionality that is exposed. When someone starts reading this class, I wouldn't expect them to know what metadata is and why it is used.